tests/prague/eip6206_jumpf/contracts.py

```diff
@@ -0,0 +1,244 @@
+                                outputs=target_outputs,
+                                height=max(target_inputs, target_outputs),
+                            ),
+                        ],
+                    )
+
+                    if stack_diff == current_outputs - target_outputs:
+                        valid.append(container)
+                    else:
+                        container.validity_error = EOFException.UNDEFINED_EXCEPTION
+                        invalid.append(container)
+
+    return (valid, invalid)
+
+
+def generate_jumpf_stack_non_returning_rules():
+    """
+    Generate tests for JUMPF where we are testing the stack rules.  Non-returning section cases.
+    """
+    valid = []
+    invalid = []
+    for stack_height in [0, 2, 4]:
+        for target_inputs in [0, 2, 4]:
+            container = Container(
+                name="stack-non-retuning_h-%d_ti-%d" % (stack_height, target_inputs),
+                sections=[
+                    quick_code(Op.JUMPF[1]),
+                    quick_code(
+                        Op.PUSH0 * stack_height + Op.JUMPF[2],
+                        height=stack_height,
+                    ),
+                    quick_code(
+                        Op.POP * target_inputs + Op.SSTORE(0, 1) + Op.STOP,
+                        inputs=target_inputs,
+                        height=max(2, target_inputs),
+                    ),
+                ],
+            )
+
+            if stack_height >= target_inputs:
+                valid.append(container)
+            else:
+                container.validity_error = EOFException.UNDEFINED_EXCEPTION
+                invalid.append(container)
+
+    return (valid, invalid)
+
+
+jump_forward = Container(
+    name="jump_forward",
+    sections=[quick_code(Op.JUMPF[1]), quick_code(Op.SSTORE(0, 1) + Op.STOP, height=2)],
+)
+jump_backward = Container(
+    name="jump_backward",
+    sections=[
+        quick_code(Op.CALLF[2] + Op.SSTORE(0, 1) + Op.STOP, height=2),
+        quick_code(Op.RETF, outputs=0),
+        quick_code(Op.JUMPF[1], outputs=0),
+    ],
+)
+jump_to_self = Container(
+    name="jump_to_self",
+    sections=[
+        quick_code(
+            Op.SLOAD(0) + Op.ISZERO + Op.RJUMPI[1] + Op.STOP + Op.SSTORE(0, 1) + Op.JUMPF[0],
+            height=2,
+        )
+    ],
+)
+jump_too_large = Container(
+    name="jump_too_large",
+    sections=[quick_code(Op.JUMPF[1025])],
+    validity_error=EOFException.UNDEFINED_EXCEPTION,
+)
+jump_way_too_large = Container(
+    name="jump_way_too_large",
+    sections=[quick_code(Op.JUMPF[0xFFFF])],
+    validity_error=EOFException.UNDEFINED_EXCEPTION,
+)
+jump_non_existent_section = Container(
+    name="jump_non_existent_section",
+    sections=[quick_code(Op.JUMPF[5])],
+    validity_error=EOFException.UNDEFINED_EXCEPTION,
+)
+callf_non_returning = Container(
+    name="callf_non_returning",
+    sections=[quick_code(Op.CALLF[1]), quick_code(Op.STOP, outputs=NON_RETURNING_SECTION)],
```

Comment:
are you sure we need to hide standard interface even more? 
in tests its oke to write explicit 
```
Section.Code(
                        code=Op.PUSH1(3) + Op.JUMP + Op.JUMPDEST + Op.STOP,
                        code_inputs=0,
                        code_outputs=NON_RETURNING_SECTION,
                        max_stack_height=1,
                    ),
```